app/src/main/java/com/zulip/android/activities/RecyclerMessageAdapter.java

```diff
@@ -381,10 +382,11 @@ public void onBindViewHolder(final RecyclerView.ViewHolder holder, final int pos
                             .setOnClickListener(new View.OnClickListener() {
                                 @Override
                                 public void onClick(View view) {
-                                    Intent i = new Intent(Intent.ACTION_VIEW);
+                                    Intent i = new Intent(zulipApp.getApplicationContext(), PhotoViewActivity.class);
                                     i.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
-                                    i.setData(Uri.parse(url));
+                                    i.putExtra(Intent.EXTRA_TEXT , url);
                                     zulipApp.startActivity(i);
+                                    ((Activity) context).overridePendingTransition(android.R.anim.fade_in, android.R.anim.fade_out);
```

Comment:
@saketkumar95 yeah such complex not needed, thats what I said above ` if its not a big hassle` just a pop up or slide in will do!